Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -289,20 +289,18 @@ static bool colorIsChallengingToHighlight(const Color& color)
 
 static bool styleIsChallengingToHighlight(const RenderStyle& style)
 {
-    auto fillPaintType = style.fill().type;
+    auto& fillPaint = style.fill();
 
-    if (fillPaintType == Style::SVGPaintType::None) {
-        auto strokePaintType = style.stroke().type;
-        if (strokePaintType != Style::SVGPaintType::RGBColor && strokePaintType != Style::SVGPaintType::CurrentColor)
-            return false;
-
-        return colorIsChallengingToHighlight(style.colorResolvingCurrentColor(style.stroke().color));
+    if (fillPaint.isNone()) {
+        if (auto strokeColor = style.stroke().tryColor())
+            return colorIsChallengingToHighlight(style.colorResolvingCurrentColor(*strokeColor));
+        return false;
     }
```

Comment:
What do you think of this alternative?
```suggestion
    auto color = (style.fill().isNone() ? style.stroke() : style.fill()).tryColor();
    return color && colorIsChallengingToHighlight(style.colorResolvingCurrentColor(*color));
```